### ISSUE TITLE
Minitest failing when running rspec after upgrading rails 4.0 to 4.1/4.2

### ISSUE THREAD

**Issue by Schermy** [seattlerb/minitest#513]
Hi there,

I've been attempting to upgrade my rails 4.0 app to rails 4.2 (via 4.1). Upgrading the app seems to work, but I cannot for the life of me get rspec to run. I'm posting this here, because during the upgrade, minitest is updated from a 4.x version to a 5.x version and my current thinking is that this is what is causing the issue.

Below is the stack trace when I run rspec. Does anyone have any insights? I'm happy to provide more details if need be. If anyone thinks that this issue is related to another library, please let me know.

	coverage report generated for RSpec to /Users/jake/Source/Rails/ny-booking/coverage. 1247 / 1365 LOC (91.36%) covered.
	/Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/minitest-5.5.1/lib/minitest/test.rb:10:in `<module:Minitest>': superclass mismatch for class Test (TypeError)
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/minitest-5.5.1/lib/minitest/test.rb:3:in `<top (required)>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `block in require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:232:in `load_dependency'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/minitest-5.5.1/lib/minitest.rb:780:in `<top (required)>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `block in require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:232:in `load_dependency'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/test_case.rb:2:in `<top (required)>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/actionpack-4.1.0/lib/action_controller/test_case.rb:428:in `<module:ActionController>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/actionpack-4.1.0/lib/action_controller/test_case.rb:6:in `<top (required)>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-rails-3.2.1/lib/rspec/rails/example/controller_example_group.rb:7:in `<module:ControllerExampleGroup>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-rails-3.2.1/lib/rspec/rails/example/controller_example_group.rb:4:in `<module:Rails>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-rails-3.2.1/lib/rspec/rails/example/controller_example_group.rb:2:in `<module:RSpec>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-rails-3.2.1/lib/rspec/rails/example/controller_example_group.rb:1:in `<top (required)>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `block in require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:232:in `load_dependency'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-rails-3.2.1/lib/rspec/rails/example.rb:2:in `<top (required)>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `block in require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:232:in `load_dependency'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-rails-3.2.1/lib/rspec/rails.rb:12:in `<top (required)>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `block in require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:232:in `load_dependency'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/activesupport-4.1.0/lib/active_support/dependencies.rb:247:in `require'
		from /Users/jake/Source/Rails/ny-booking/spec/spec_helper.rb:9:in `<top (required)>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/configuration.rb:1181:in `require'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/configuration.rb:1181:in `block in requires='
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/configuration.rb:1181:in `each'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/configuration.rb:1181:in `requires='
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/configuration_options.rb:110:in `block in process_options_into'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/configuration_options.rb:109:in `each'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/configuration_options.rb:109:in `process_options_into'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/configuration_options.rb:22:in `configure'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/runner.rb:96:in `setup'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/runner.rb:85:in `run'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/runner.rb:70:in `run'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/lib/rspec/core/runner.rb:38:in `invoke'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/gems/rspec-core-3.2.1/exe/rspec:4:in `<top (required)>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/bin/rspec:23:in `load'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/bin/rspec:23:in `<main>'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/bin/ruby_executable_hooks:15:in `eval'
		from /Users/jake/.rvm/gems/ruby-2.2.1@ny-booking/bin/ruby_executable_hooks:15:in `<main>'

**Comment by zenspider** [comment]
I highly doubt this has anything to do with minitest and is a bad interaction on your rails upgrade.

But to figure it out??? \**shrug*\*

It really depends on whatever that other superclass of `Minitest::Test` is. If you can bring up the rails console (aka irb) maybe you can load the code in question and try to reproduce the problem... then look at where it thinks the superclass is coming from.

Hell... hack in before `minitest-5.5.1/lib/minitest/test.rb:3` and print out the relevant info if you have to.

**Comment by Schermy** [comment]
@zenspider What do you mean by a bad interaction? I doubt that I've upgraded incorrectly since I've attempted this a few times now with the same result.

Part of the issue is that I don't particularly understand minitest or its relationship to rspec and rails all that well which does tend to make debugging the issue harder. Can you shed any light on this @zenspider?

**Comment by zenspider** [comment]
Focus on this:

> _"It really depends on whatever that other superclass of Minitest::Test is"_

Get that superclass in order to figure out who's breaking things.

**Comment by zenspider** [comment]
Another approach, flip the error around to find the culprit. Put the correct skeleton at the top of your spec helper:

```ruby
module Minitest
  class Runnable; end
  class Test < Runnable; end
end
```

run rspec, and see what blows.

**Comment by Schermy** [comment]
Error seems to have been caused by the time-warp gem. When we excluded it from our build, rspec ran just fine.

**Comment by tenderlove** [comment]
Looks like there was a bug in the timewarp gem that they fixed here: time-warp@b55161913aec9720278939e6e3b015b37a82e41e

I'm not sure if it's been released yet, but maybe try upgrading the gem?  Thanks for reporting this!
